Include/internal/pegen_interface.h

```diff
@@ -1,4 +1,3 @@
-#ifndef Py_LIMITED_API
 #ifndef Py_PEGENINTERFACE
 #define Py_PEGENINTERFACE
```

Comment:
Oh, it's even better if the header file can be marked as internal!

I suggest to add the following guard (see for example pycore_pymem.h):

```
#ifndef Py_BUILD_CORE
#  error "this header requires Py_BUILD_CORE define"
#endif
```